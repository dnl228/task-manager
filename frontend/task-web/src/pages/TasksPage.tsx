import Header from '../components/Header';
import FilterBar from '../components/FilterBar';
import TaskForm from '../components/TaskForm';
import TaskList from '../components/TaskList';

export default function TasksPage() {
  return (
    <div style={{ maxWidth: 720, margin: '2rem auto', padding: 16 }}>
      <Header />
      <FilterBar />
      <TaskForm />
      <TaskList />
    </div>
  );
}
