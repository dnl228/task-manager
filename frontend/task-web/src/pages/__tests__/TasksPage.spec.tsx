import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import TasksPage from '../TasksPage';

jest.mock('../../components/Header', () => ({ __esModule: true, default: () => <header>HeaderMock</header> }));
jest.mock('../../components/FilterBar', () => ({ __esModule: true, default: () => <div>FilterBarMock</div> }));
jest.mock('../../components/TaskForm', () => ({ __esModule: true, default: () => <form>TaskFormMock</form> }));
jest.mock('../..//components/TaskList', () => ({ __esModule: true, default: () => <ul>TaskListMock</ul> }));

describe('TasksPage', () => {
  afterEach(() => jest.resetAllMocks());

  test('matches snapshot', () => {
    const { asFragment } = render(<TasksPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});