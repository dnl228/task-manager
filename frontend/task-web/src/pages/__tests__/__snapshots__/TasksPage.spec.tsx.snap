// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TasksPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="max-width: 720px; margin: 2rem auto; padding: 16px;"
  >
    <header>
      HeaderMock
    </header>
    <div>
      FilterBarMock
    </div>
    <form>
      TaskFormMock
    </form>
    <ul>
      TaskListMock
    </ul>
  </div>
</DocumentFragment>
`;
